### DISCUSSION TITLE
Cmdline cmp does not know to exclude search modifiers; And: How to limit completion popup to show after typing N characters first

### DISCUSSION THREAD

**Discussion by unphased** [hrsh7th/nvim-cmp#1474] [Q&A]
## Issue 1
I have `vim.keymap.set({ "n", "v" }, "/", "/\\V")` and this makes searches start with `/\V` (very nomagic mode). Unfortunately this clashes with 
```
cmp.setup.cmdline({ '/', '?' }, {
  mapping = cmp.mapping.preset.cmdline(),
  sources = {
    { name = 'buffer' }
  }
})
```
because cmp tries to match with the leading v. 

## Issue 2
There are edge cases when the popup is just too huge and covers up too much buffer. I'd prefer not to limit the number of rows the completion menu could display, I'd just like it not to run or appear until I've typed a configurable number of chars in, for the `/`/`?` cmdline completion. I hope that is configurable. I did some searching but did not find the name of this option.

**Comment by fitrh** [comment]
For the second issue, you can use `completion.keyword_length` 


```lua 
cmp.setup.cmdline(":", {
  completion = {
    keyword_length = 5,
  },
  sources = {
    { name = "cmdline" },
    { name = "path" },
  },
})
```
